Lib/test/test_tools/test_pathfix.py

```diff
@@ -44,6 +48,22 @@ def pathfix(self, shebang, pathfix_flags, exitcode=0, stdout='', stderr=''):
 
         return new_shebang
 
+    def test_recursive(self):
+        self.temp_directory = support.TESTFN + '.d'
+        self.addCleanup(support.rmtree, self.temp_directory)
+        os.mkdir(self.temp_directory)
+        self.temp_file = self.temp_directory + os.sep + \
+                         os.path.basename(support.TESTFN) + '-t.py'
+        temp_directory_basename = os.path.basename(self.temp_directory)
+        expected_stderr = f'recursedown(\'{temp_directory_basename}\')\n'
```

Comment:
I suggest:

```
tmpdir = support.TESTFN
self.addCleanup(support.rmtree, tmpdir)
os.mkdir(tmpdir)
expected_stderr = f"recursedown('{os.path.basename(tmpdir)}')\n"
... self.pathfix(..., directory=tmpdir, ...) ...
```

By the way, the self.temp_file attribute is useless. You can simply remove the setUp() method. If you replace pathfix() `filename` parameter with a new `directory` parameter, you can write something like:

```
if directory:
  pathfix_arg = directory
  filename = os.path.join(directory, 'script.py')
else:
  filename = support.TESTFN
  self.addCleanup(support.unlink, filename) 
  pathfix_arg = filename

... subprocess.run([sys.executable, self.script, *pathfix_flags, '-n', pathfix_arg], ...) ...
```

Reply:
Wouldn't adding the self.addCleanup call to pathfix cause the cleanup to be added multiple times (each time we call pathfix)? And if yes, is this really desired?
When are the cleanups run? After each single test or after all tests of this class have been executed?

Reply:
Right. One solution is to use try/finally to manually call support.unlink(filename) if directory is not set.

Reply:
Any reason why you don't like the setUp method and the addCleanup approach and prefer a local try / finally solution?

Reply:
Well, you can keep setUp() as it is if you prefer.